src/nomad_simulations/outputs.py

```diff
@@ -140,23 +105,52 @@ class SCFOutputs(Outputs):
-        description="""
-        Flag indicating whether the output property is converged or not after a SCF process. This quantity is connected
-        with `SelfConsistency` defined in the `numerical_settings.py` module.
-        """,
-    )
-
-    self_consistency_ref = Quantity(
-        type=SelfConsistency,
-        description="""
-        Reference to the `SelfConsistency` section that defines the numerical settings to converge the
-        output property.
-        """,
-    )
+    def resolve_is_derived(self, logger: BoundLogger) -> bool:
+        """
+        Resolves if the physical property is converged or not after a SCF process.
 
-    # TODO add more functionality to automatically check convergence from `self_consistency_ref` and the last two `scf_steps`
+        Returns:
+            (bool): The flag indicating whether the physical property is converged or not after a SCF process.
+        """
+        for property_name in self.m_def.all_sub_sections.keys():  # ! Check this
+            # We get the property of `SCFOutputs` that we want to check
+            prop = getattr(self, property_name)
+
+            # If it is None or does not contain a `self_consistency_ref` quantity we continue
+            if prop is None:
+                continue
+            scf_parameters = prop.self_consistency_ref
+            if scf_parameters is None:
+                continue
+
+            # We check the last 2 `scf_steps` values of such property
+            scf_steps_convergence = self.scf_steps[:-2]
+            scf_props = []
+            for scf_step in scf_steps_convergence:
+                scf_prop = getattr(scf_step, property_name)
+                if scf_prop is None:
+                    logger.warning(f'The `scf_step.{property_name}` does not exist.')
+                    break
+                if scf_prop.value.unit != scf_parameters.threshold_unit:
+                    logger.error(
+                        f'The units of the `scf_step.{property_name}.value` does not coincide with the units of the `self_consistency_ref.threshold_unit`.'
+                    )
+                    return
+                scf_props.append(scf_prop.value.magnitude)  # we use only the magnitude
+
+            # We check if the property `is_scf_converged` checking its difference with respect to `self_consistency_ref.threshold_change`
+            threshold = scf_parameters.threshold_change
+            scf_diff = abs(scf_props[0] - scf_props[1])
+            if scf_diff <= threshold:
+                prop.is_scf_converged = True
+            else:
+                logger.info(
+                    f'The SCF process for the property `{property_name}` did not converge.'
+                )
+                prop.is_scf_converged = False
 
     def normalize(self, archive, logger) -> None:
         super().normalize(archive, logger)
+
+        # Resolve the `is_scf_converged` flag for all SCF obtained properties
+        self.resolve_is_derived(logger)
```

Comment:
I would only run this after checking that `is_derived` hasn't been set: typically, the parser can extract convergence.